Assets/Project/Modules/Enemies/GeneralEnemyScripts/VFXRelated/EnemyVisuals.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Cysharp.Threading.Tasks;
+using UnityEngine;
+
+public class EnemyVisuals : MonoBehaviour
+{
+    //This is the base script, any unique additions to these
+    //effects must be done by creating a new unique script
+    //that inherits from this one.
+    
+    [System.Serializable]
+    public class OriginalMeshData
+    {
+        public MeshRenderer _mesh;
+        [HideInInspector] public Material _originalMaterial;
+    }
+
+    [SerializeField] private GeneralEnemyVFXConfig _visualConfig;
+    [SerializeField] [Tooltip("First mesh on the list will be the one that gets 'hurt'")]
+    private List<OriginalMeshData> _originalMeshDatas = new();
+
+    private void Awake()
+    {
+        foreach (var data in _originalMeshDatas)
+        {
+            data._originalMaterial = data._mesh.material;
+        }
+    }
+
+    public virtual void OnHitEffects(float healthCoef)
```

Comment:
I wouldn't name this methods starting with "On", but rather "Play" (PlayHitEffects), the same with the other methods (PlayFlashEffect, PlayDeathEffects).
I think we should reserve the naming of methods that start with "On" for multi-purpouse functions (like the ones a mediator may have).
Also, "OnHitEffects" isn't really descriptive, and it sounds like a reaction to when a HitEffects that was just played. "PlayHitEffects" on the other hand, is not ambiguous.